src/settings/SettingsStore.ts

```diff
@@ -334,6 +336,17 @@ export default class SettingsStore {
         return !SETTINGS[settingName].controller?.settingDisabled ?? true;
     }
 
+    /**
+     * Retrieves the reason a setting is disabled if one is assigned.
+     * If a setting is not disabled this will return undefined.
```

Comment:
```suggestion
     * If a setting is not disabled, or no reason is given by the `SettingController`,
     * this will return undefined.
```

Reply:
Was implied by 

> if one is assigned.

but no harm in being explicit